tests/tests_unit/test_config_set.py

```diff
@@ -0,0 +1,73 @@
+from unittest.mock import patch
+
+import pytest
+from packaging.requirements import Requirement
+
+from _nebari.config_set import ConfigSetMetadata, read_config_set
+
+test_version = "2024.12.2"
+
+
+def test_valid_version_requirement():
```

Comment:
Can we parameterize this test with a few different version strings to ensure it works in all the allowed formats? ideally including pre-release versions.

Reply:
done